bin/patchCluster.sh

```diff
@@ -0,0 +1,251 @@
+    echo -e ""
+    echo -e "Examples: \n"
+    echo -e "       ./patchCluster.sh -s reqmgr2 11270 12120 \n"
+    echo -e "       ./patchCluster.sh -p pod/reqmgr2-bcdccd8c6-hsmlj -f /tmp/11270.patch \n"
+    echo -e "       git diff --no-color | ./patchCluster.sh -s reqmgr2 -n \n"
+    echo -e "       curl https://patch-diff.githubusercontent.com/raw/dmwm/WMCore/pull/11270.patch | ./patchCluster.sh -s reqmgr2 \n"
+
+}
+
+# Set defaults
+currPods=""
+currService=""
+currDeployment=""
+zeroOnly=false
+zeroCodeBase=true
+extPatchFile=""
+while getopts ":f:p:s:d:znh" opt; do
+    case ${opt} in
+        p)
+            currPods=$OPTARG
+            ;;
+        s)
+            currService=$OPTARG
+            ;;
+        d)
+            currDeployment=$OPTARG
+            ;;
+        f)
+            extPatchFile=$OPTARG
+            ;;
+        z)
+            zeroOnly=true
+            ;;
+        n)
+            zeroCodeBase=false
+            ;;
+        h)
+            usage
+            exit 0 ;;
+        \? )
+            echo -e "\nERROR: Invalid Option: -$OPTARG\n"
+            usage
+            exit 1 ;;
+        : )
+            echo -e "\nERROR: Invalid Option: -$OPTARG requires an argument\n"
+            usage
+            exit 1 ;;
+    esac
+done
+
+# shift to the last  parsed option, so we can consume the patchList with a regular shift
+shift $(expr $OPTIND - 1 )
+
+
+# if fd 0 (stdin) is open and refers to a terminal - then we are running the script directly, without a pipe
+# if fd 0 (stdin) is open but does not refer to the terminal - then we are running the script through a pipe
+if [ -t 0 ] ; then pipe=false; else pipe=true ; fi
+
+patchList=$*
+
+currCluster=`kubectl config get-clusters |grep -v NAME`
+nameSpace=dmwm
+
+echo
+echo
+echo
+echo ========================================================
+echo "INFO: CLUSTER: $currCluster"
+echo --------------------------------------------------------
+
+if $zeroOnly; then
+    echo "INFO: Only Zeroing the code base No patches will be applied"
+    echo "INFO: Source files affected to be taken from patches: $patchList"
+elif $pipe ;then
+    echo "INFO: Patching from StdIn"
+elif [[ -n $extPatchFile ]]; then
+    echo "INFO: Patching from file: $extPatchFile"
+elif [[ -z $patchList ]] ; then
+    echo "ERROR: Neither patchList provided nor patchFile nor patching from StdIn"
+    exit
+else
+    echo "INFO: Applying patches: $patchList"
+fi
+
+
+# -----------------------------------------------------------
+# Build patchComponent.sh script command to be executed at the pod:
+
+# NOTE: We do not support patching from file and patching from command line simultaneously
+#       If both provided at the command line patching from command line takes precedence
+
+podCmdActions="wget https://raw.githubusercontent.com/todor-ivanov/WMCoreAux/master/bin/patchComponent.sh -O /data/patchComponent.sh && chmod 755 /data/patchComponent.sh "
```

Comment:
Indeed, `wget`  is a requirement.  I can try to double check if it is present at the container, but this would make the `podCmd*` commands executions more complicate.... - would require two extra steps from the caller script, so at least one more cycle there.... The idea here is that they should go in as "one-liners" because they would be pushed through the kubectl connection session as a single command with accumulated errors. So the only way to achieve that is through an `&&` logical chain.  I'll see what I can do, though, and will write back the results from my tests here.

Reply:
so:
> I think you should use back quote to get output of command to assign it to a shell variable. please check

Actually not. Because we  need to delay the execution as much as possible so it happens at the container. A back quote here would push the execution to happen at the caller's shell. So we better first construct the strings and then send it through the session to the containers.